ftag/sample.py

```diff
@@ -16,6 +16,9 @@ class Sample:
     weights: list[float] | None = None
```

Comment:
A cleaner way of doing this might be to change the type signature here, and then cast the weights to a tuple before passing them to the `Sample` constructor
```suggestion
    weights: tuple[float] | None = None
```